src/runtime/pyscope.cs

```diff
@@ -0,0 +1,427 @@
+                    if (op == IntPtr.Zero)
+                    {
+                        throw new PythonException();
+                    }
+                    if (op == Runtime.PyNone)
+                    {
+                        return null;
+                    }
+                    return new PyObject(op);
+                }
+                else
+                {
+                    throw new PyScopeException(String.Format("'ScopeStorage' object has no attribute '{0}'", name));
+                }
+            }
+        }
+
+        /// <summary>
+        /// TryGetVariable Method
+        /// </summary>
+        /// <remarks>
+        /// Returns the value of the variable, local variable first.
+        /// If the variable is not exists, return null.
+        /// </remarks>
+        public bool TryGetVariable(string name, out PyObject value)
+        {
+            Check();
+            using (var pyKey = new PyString(name))
+            {
+                if (Runtime.PyMapping_HasKey(variables, pyKey.obj) != 0)
+                {
+                    IntPtr op = Runtime.PyObject_GetItem(variables, pyKey.obj);
+                    if (op == IntPtr.Zero)
+                    {
+                        throw new PythonException();
+                    }
+                    if (op == Runtime.PyNone)
+                    {
+                        value = null;
+                        return true;
+                    }
+                    value = new PyObject(op);
+                    return true;
+                }
+                else
+                {
+                    value = null;
+                    return false;
+                }
+            }
+        }
+
+        public T GetVariable<T>(string name)
+        {
+            Check();
+            PyObject pyObj = GetVariable(name);
+            if (pyObj == null)
+            {
+                return default(T);
+            }
+            return (T)ToManagedObject<T>(pyObj);
+        }
+
+        public bool TryGetVariable<T>(string name, out T value)
+        {
+            Check();
+            PyObject pyObj;
+            var result = TryGetVariable(name, out pyObj);
+            if (!result)
+            {
+                value = default(T);
+                return false;
+            }
+            if (pyObj == null)
+            {
+                value = default(T);
+                return true;
+            }
+            value = (T)ToManagedObject<T>(pyObj);
+            return true;
+        }
+
+        public override bool TryGetMember(GetMemberBinder binder, out object result)
+        {
+            result = this.GetVariable(binder.Name);
+            return true;
+        }
+
+        public override bool TrySetMember(SetMemberBinder binder, object value)
+        {
+            this.SetVariable(binder.Name, value);
+            return true;
+        }
+
+        private object ToManagedObject<T>(PyObject pyObj)
```

Comment:
I cannot agree more.
I thought the name AsManagedObject is too verbose because this method is heavily used, what's more, it can not accept object/dynamic as it parameter (it can be fixed easily).

 I had planed to replace it for a long time. Since you mentioned it now, a new As<T> method was added to the PyObject.